ppr-ui/src/utils/registration-helper.ts

```diff
@@ -76,3 +76,100 @@ export async function saveFinancingStatementDraft (stateModel:StateModelIF): Pro
   }
   return draftResponse
 }
+
+/** Save new financing statement. Data to be saved is in the store state model. */
+export async function saveFinancingStatement (stateModel:StateModelIF): Promise<FinancingStatementIF> {
+  const registrationType: RegistrationTypeIF = stateModel.registrationType
+  var error:ErrorIF = null
+  var draft:DraftIF = stateModel.draft
+  const trustLength = stateModel.lengthTrustStep
+  const parties:AddPartiesIF = stateModel.addSecuredPartiesAndDebtorsStep
+  const collateral:AddCollateralIF = stateModel.addCollateralStep
+  var statement:FinancingStatementIF = {
+    type: stateModel.registrationType.registrationTypeAPI,
+    lifeInfinite: trustLength.lifeInfinite,
+    lifeYears: trustLength.lifeYears,
+    registeringParty: parties.registeringParty,
+    securedParties: parties.securedParties,
+    debtors: parties.debtors,
+    vehicleCollateral: collateral.vehicleCollateral,
+    generalCollateral: [],
+    clientReferenceId: stateModel.folioOrReferenceNumber
+  }
+  if (draft !== null && draft.financingStatement !== null) {
+    statement.documentId = draft.financingStatement.documentId
+  }
+  if (statement.type === 'SA') {
+    statement.trustIndenture = trustLength.trustIndenture
+  }
+  if (collateral.generalCollateral !== null && collateral.generalCollateral !== '') {
+    var generalCollateral: GeneralCollateralIF = { description: collateral.generalCollateral }
+    statement.generalCollateral = [generalCollateral]
+  }
+  // Now tidy up, deleting objects that are empty strings to pass validation.
+  // For example, party.birthDate = '' will fail validation.
+  statement.registeringParty = cleanupParty(statement.registeringParty)
+  for (let i = 0; i < statement.debtors.length; i++) {
+    statement.debtors[i] = cleanupParty(statement.debtors[i])
+  }
+  for (let i = 0; i < statement.securedParties.length; i++) {
+    statement.securedParties[i] = cleanupParty(statement.securedParties[i])
+  }
+  if (statement.vehicleCollateral !== null) {
+    for (let i = 0; i < statement.vehicleCollateral.length; i++) {
+      if (statement.vehicleCollateral[i].year !== null) {
+        if (statement.vehicleCollateral[i].year === '') {
+          delete statement.vehicleCollateral[i].year
+        } else if (typeof statement.vehicleCollateral[i].year === 'string') {
+          statement.vehicleCollateral[i].year = Number(statement.vehicleCollateral[i].year)
+        }
+      }
+    }
+  }
+  // console.log('Save Financing Statement: ' + JSON.stringify(statement))
+  // Now save the financing statement.
+  const apiResponse = await createFinancingStatement(statement)
+
+  if (apiResponse !== undefined && apiResponse.error === undefined) {
+    console.log('saveFinancingStatement successful for registration number ' + apiResponse.baseRegistrationNumber)
```

Comment:
probably didn't mean to leave in this log or the commented out one above?

Reply:
Thanks log statements removed.